### DISCUSSION TITLE
Loading different GLB file based on ar-mode

### DISCUSSION THREAD

**Discussion by uprealdigital** [google/model-viewer#2977] [Q&A] [answered]
Hi,

Since I can add labels with WebXR and not with SceneViewer and Quick-Look, I would like to change the .glb model used based on the ar-mode used. 

Example:

WebXR: catNoLabels.glb (Labels generated with Hotspots)
SceneViewer: catWithLabels.glb (Labels embedded in the 3D model)
Quick-Look: catWithLabels.glb (Labels embedded in the 3D model && I'm using the auto-generated USDZ in that case based on the specified GLB 3D object)

Is that something that's possible?

Cheers,

**Comment by samaneh-kazemi** [comment]
Hi @ARportals , I'm not sure what is the question. Can you share a snippet of code on what you mean?

**Reply by uprealdigital** [reply]
@samaneh-kazemi 

I'm not sure how I would describe my problem with a code snippet.

Basically, ModelViewer has an attribute called ar-modes where I usually specify the following value:

ar-modes="WebXR,SceneViewer,Quick-Look"

It will try WebXR first, if that fails it will try SceneViewer and if that fails it will try Quick-look

But that means that we don't know in advance which ar-mode it will use for a particular device.

And each ar-mode is different like WebXR can have HTML hotspots and SceneViewer or Quicklook can't.

So I want to load a different .glb model based on the ar-mode used.

Example:

WebXR => cats.glb
Quicklook => dogs.glb
SceneViewer => bird.glb

Since each ar-mode reacts differently, it makes total sense to switch the glb based on the ar-mode.

**Reply by zuggamasta** [reply]
Sorry for digging out this old question, but I have the same issue I need to do a minor change to my model to make it look good in scene viewer/webXR du to the materias.

is it possible to either send another model OR other model variant?

Thank you very much in advance.

**Comment by zuggamasta** [comment, marked as answer]
Hey there,

maybe this will be helpful for someone in the future: If you do not need a different mesh, but a different Material on the Mesh you can switch the material when entering AR mode.



```
    const modelViewerVariants = document.querySelector("model-viewer#model");
    modelViewerVariants.variantName = 'BrowserMaterials'

    const arButton = document.querySelector("#ar-button");

    arButton.addEventListener("click", function() {
        modelViewerVariants.variantName = 'ArMaterils'

    });
   ```

**Reply by elalish** [reply]
This will only work for autogenerated USDZ for iOS Quick Look. WebXR uses the same renderer as 3D, so there should be no change needed there anyway. This approach won't work for Scene Viewer, but hopefully that' not needed as WebXR is available almost everywhere Scene Viewer is.
